src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java

```diff
@@ -17,4 +24,27 @@
         String translated = translateToLepConvention(profile);
         return new String[]{translated};
     }
+
+    @Override
+    protected <T> T getRequiredParam(LepMethod method, String paramName, Class<T> valueType) {
+        return ofNullable(getParamValue(method, paramName, valueType))
+            .or(() -> getHeaderValue(paramName, valueType))
+            .orElseThrow(() -> new IllegalArgumentException(
+                String.format("Neither LEP method %s nor request header contain parameter %s",
+                    getMethodDescription(method), paramName)));
+    }
+
+    private <T> Optional<T> getHeaderValue(final String headerName, final Class<T> valueType) {
+        return ofNullable((ServletRequestAttributes) RequestContextHolder.getRequestAttributes())
```

Comment:
please consider rewriting in a functional way + avoid potential NPE in attributes.getRequest():
```
return ofNullable(RequestContextHolder.getRequestAttributes())
            .map(ServletRequestAttributes.class::cast)
            .map(ServletRequestAttributes::getRequest)
            .map(request -> request.getHeader(headerName))
            .map(valueType::cast);
```

Reply:
Done